envoy/upstream/cluster_manager.h

```diff
@@ -150,7 +150,6 @@ struct ClusterConnectivityState {
   ~ClusterConnectivityState() {
     ASSERT(pending_streams_ == 0);
     ASSERT(active_streams_ == 0);
-    ASSERT(connecting_and_connected_stream_capacity_ == 0);
```

Comment:
Hm. I don't think I understand why this is being removed. Can you say more?

Reply:
Ah, maybe I misunderstood your last comment, I thought you wanted me to remove that assertion, I'll bring it back, and I'll revert the Http3ConnPoolImpl change.

Reply:
Sorry! What I meant previously was that in ~Http3ConnPoolImpl() we were setting connecting_and_connected_stream_capacity_ to zero with a comment explaining that Http3ConnPoolImpl was a bit weird in how it tracked state. I wondered if that was the *only* piece of state we needed to tweak, so was suggesting we ASSERT() in Http3ConnPoolImpl that the other fields are already 0. But looking at this code here, I see that we're already going that.... just in the base class. (Presumably it was this assert which motivated that change)

So I think your original change was fine. Perhaps add a sentence to the comment which remarks that the base class destructor verifies that the other fields have been cleaned up?